src/components/PersonLink.tsx

```diff
@@ -0,0 +1,95 @@
+import {
+  Link,
+  useNavigate,
+  useParams,
+  useSearchParams,
+} from 'react-router-dom';
+import { Person } from '../types';
+import { useEffect, useRef } from 'react';
+
+type Props = {
+  person: Person;
+  people: Person[];
+};
+
+export const PersonLink: React.FC<Props> = ({ person, people }) => {
+  const { slug } = useParams();
+  const personRef = useRef(null);
```

Comment:
The `personRef` is initialized with `null`, which is correct. However, ensure that the ref is correctly assigned to the DOM element to avoid potential null reference errors.